src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -724,18 +715,9 @@ public function appendToFile($filename, $content)
         }
     }
 
-    /**
-     * @param mixed $files
-     *
-     * @return \Traversable
-     */
-    private function toIterator($files)
+    private function toIterable($files): iterable
     {
-        if (!$files instanceof \Traversable) {
-            $files = new \ArrayObject(is_array($files) ? $files : array($files));
-        }
-
-        return $files;
+        return is_iterable($files) ? $files : array($files);
```

Comment:
this implementation is not compatible with the previous, won't it create BC breaks f the return is forwarded as return value of a public method?

Reply:
private API :) checked all usages; they truly rely on a `foreach`able value.